modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridClockImpl.java

```diff
@@ -86,8 +74,8 @@ public long nowLong() {
     }
 
     @Override
-    public long currentLong() {
-        long current = currentTime();
```

Comment:
The main goal of adding "final" is to allow behavior modification only by overloading physicalTime method (for test purposes)
What does it mean: same code is used at least thrice ? which code ?

Reply:
`physicalTime() << LOGICAL_TIME_BITS_SIZE` is used 3 times in this class. How about restoring the `currentTime()` method, but making it private? Like this

```
    private long currentTime() {
        return physicalTime() << LOGICAL_TIME_BITS_SIZE;
    }
```

This will still allow to use `physicalTime()` (overridden by test clock), but reduce duplication (and `currentTime()` will not be overridable as it's private).

Reply:
🆗